launchdarkly/ai_model_config_helper.go

```diff
@@ -0,0 +1,185 @@
+package launchdarkly
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	ldapi "github.com/launchdarkly/api-client-go/v17"
+)
+
+func baseAIModelConfigSchema(isDataSource bool) map[string]*schema.Schema {
+	schemaMap := map[string]*schema.Schema{
+		PROJECT_KEY: {
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			Description:      addForceNewDescription("The AI model config's project key. A change in this field will force the destruction of the existing resource and the creation of a new one.", !isDataSource),
+			ValidateDiagFunc: validateKey(),
+		},
+		KEY: {
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			ValidateDiagFunc: validateKey(),
+			Description:      addForceNewDescription("The unique key that references the AI model config. A change in this field will force the destruction of the existing resource and the creation of a new one.", !isDataSource),
+		},
+		NAME: {
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("The human-friendly name for the AI model config.", !isDataSource),
+		},
+		ID: {
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
```

Comment:
does the user assign this or does LD? if LD assigns this it should always be computed and the force new is moot